src/V3Delayed.cpp

```diff
@@ -688,26 +831,24 @@ class DelayedVisitor final : public VNVisitor {
             iterateChildren(nodep);
         }
         if (m_timingDomains.empty()) return;
-        if (AstActive* const actp = VN_AS(nodep->user2p(), Active)) {
-            // Merge all timing domains (and possibly the original active's domain)
-            // to create a sentree for the pre/post logic
-            // TODO: allow multiple sentrees per active, so we don't have
-            //       to merge them and create a new trigger
-            AstSenTree* senTreep = nullptr;
-            if (actp->sensesp()->hasClocked()) senTreep = actp->sensesp()->cloneTree(false);
-            for (AstSenTree* const domainp : m_timingDomains) {
-                if (!senTreep) {
-                    senTreep = domainp->cloneTree(false);
-                } else {
-                    senTreep->addSensesp(domainp->sensesp()->cloneTree(true));
-                    senTreep->multi(true);  // Comment that it was made from multiple domains
-                }
-            }
-            V3Const::constifyExpensiveEdit(senTreep);  // Remove duplicates
-            actp->sensesp(senTreep);
-            actp->sensesStorep(senTreep);
+
+        // There were some timing domains involved in the process. Add all of them as sensitivities
+        // of all NBA targets in this process. Note this is a bit of a sledgehammer, we should only
+        // need those that directly preceed the NBA in control flow, but that is hard to compute,
+        // so we will hammer away.
+
+        // First gather all senItems
+        AstSenItem* senItemp = nullptr;
+        for (AstSenTree* const domainp : m_timingDomains) {
+            senItemp = AstNode::addNext(senItemp, domainp->sensesp()->cloneTree(true));
         }
         m_timingDomains.clear();
+        // Add them to all nba targets we gathered in this process
+        for (size_t i = firstNBAAddedIndex; i < m_nbas.size(); ++i) {
+            m_vscpInfo(m_nbas[i].vscp).addSensitivity(senItemp);
+        }
+        // Done with these
+        VL_DO_DANGLING(senItemp->deleteTree(), senItemp);
```

Comment:
```suggestion
        VL_DO_DANGLING(pushDeleteps(senItemp), senItemp);
```

To match other deleters/aid debug.